### DISCUSSION TITLE
Documentation: Dual Stack networking

### DISCUSSION THREAD

**Discussion by withinboredom** [k3s-io/k3s#6398] [Q&A]
Currently, in the k3s documentation, it says:

```
Dual-stack networking must be configured when the cluster is first created. It cannot be enabled on an existing single-stack cluster.
```

However, it does not give a reason as to why. Is this a technical limitation (as in some code makes this literally impossible), or does it just require shutting down the entire cluster and thus impractical? Or is this just something no one has ever tried before and thus unsupported?

**Comment by brandond** [comment]
The Kubernetes IPAM only allocates addresses when things are first added to the cluster - whether it's a node, service, so on. You would basically need to remove and re-create everything in your cluster to enable dual-stack operation. Since this is a lot of work, upstream doesn't test or support this, and neither do we.

**Reply by withinboredom** [reply]
Awesome! Thanks for the information!
